components/Navbar.tsx

```diff
@@ -0,0 +1,161 @@
+"use client";
+
+import { Fragment } from "react";
+import { Disclosure, Menu, Transition } from "@headlessui/react";
+import {
+  Bars3Icon,
+  XMarkIcon,
+  BuildingStorefrontIcon,
+  UserCircleIcon,
+} from "@heroicons/react/24/outline";
+import { classNames } from "../src/utils";
+import { useRouter } from "next/navigation";
+
+const navigation = [
+  { name: "Dashboard", href: "/dashboard", current: true },
+  { name: "Page #1", href: "#", current: false },
+  { name: "Page #2", href: "#", current: false },
+];
+
+export default function Navbar({ signedIn }: { signedIn: boolean }) {
+  const router = useRouter();
+
+  const handleSignOut = async () => {
+    await fetch("/auth/signout", {
+      headers: { "Content-Type": "application/json" },
+    });
```

Comment:
Something to consider:

We have a `createServerActions()` function in the nextjs auth package that will generate these server actions for us if we want to suggest that here.

So like:

```ts
// /auth-actions.ts or something like that
import { auth } from "@/edgedb";

const {
  signout,
  // etc
} = auth.createServerActions();
```

And then here:

```suggestion
    // import { signout } from "@/auth-actions";
    await signout();
```

Reply:
I wanted to use it, but it's a client component, and importing `auth` was failing (node stuff is required).

Reply:
I've updated it.